### PR TITLE
Add meson build support

### PR DESCRIPTION
**Pull request by whs**
I remembered that MATE was looking to migrate to [Meson](https://wiki.mate-desktop.org/developers-corner/wayland-meson/). So I'd like to pitch in

This tries to be mostly 1-1 port to the autotools version.

## Progress

- [x] Build eom binary
- [x] Build yelp/man
- [x] Build gobject introspection
- [x] Build plugins
- [x] Build thumbnailer
- [x] ~Build api docs~ doesn't work with current meson
- [x] Build dist
- [x] Install

## Systems tested

- [x] Arch Linux
- [x] Ubuntu 22.10

I don't have access to a Mac, and I'm not sure if it is a supported target (I see that configure have mentions to Mac)

### CODE REVIEW

**Comment by whs** [comment]
Seems that meson doesn't support linking objects with gtkdoc (it only add include_directories but not object files)

**Review by lukefromdc — COMMENTED** [review]
A test build (without building gtk-doc which I have never done) went fine. Some "could not merge translation" warnings also appear with an autotools build so probably nothing to do with this PR. No runtime issues on a test run

File: data/icons/meson.build [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+foreach size : ['16x16', '22x22', '24x24', '32x32']
+  install_data([
+    size / 'apps/eom.png',
+  ], install_dir: appiconsdir / size / 'apps')
+endforeach
+
+install_data(['scalable/apps/eom.svg'], install_dir: appiconsdir / 'scalable' / 'apps')
+
+meson.add_install_script(find_program('gtk-update-icon-cache'), '-f', '-t', appiconsdir, skip_if_destdir: true)
```
Comment (eli-schwartz) [inline data/icons/meson.build:33]: https://mesonbuild.com/Gnome-module.html#gnomepost_install

You can do `gnome.post_install(gtk_update_icon_cache: true)` and it handles both finding the program for you (under a couple different names) and invoking it depending on destdir.

Comment (eli-schwartz) [reply data/icons/meson.build:33]: Also you're already using `gnome.post_install` elsewhere in this PR. You can call it multiple times and use additional features.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: data/icons/meson.build (hunk 1 above)
Comment (whs) [reply data/icons/meson.build:33]: Good catch. Initially I targeted 0.51 so I wrote the install script myself as this requires 0.57. Then the target version moved around during development so I probably miss some potential features I could've used.

Comment (whs) [reply data/icons/meson.build]: Ok, updated

Comment (raveit65) [reply data/icons/meson.build]: @eli-schwartz can we resolve the conversation?

**Comment by raveit65** [comment]
@whs 
Please rebase the branch.
```
[rave@mother eom]$ git checkout -b whs-master master
Switched to a new branch 'whs-master'
[rave@mother eom]$ git pull https://github.com/whs/eom.git master
From https://github.com/whs/eom
 * branch            master     -> FETCH_HEAD
fatal: Not possible to fast-forward, aborting.
```

File: data/icons/meson.build (hunk 1 above)
Comment (eli-schwartz) [reply data/icons/meson.build]: Indeed we can, my suggestion was implemented.

**Comment by whs** [comment]
Rebased, seems that the build script do not change much in the mean time other than ImageMagick 6 detection. So no additional changes.

**Review by raveit65 — APPROVED** [review]
LGTM

**Review by lukefromdc — APPROVED** [review]
Building, installingh, and uninstalling with EOM not previouslly installed went fine. If EOM built w autotools is already installed, we get an install error on replacing a file with a symlink, but this may just be meson building with different path options (I have a lot of specification for path in my autotools builds for multilib etc)..

That won't affect distro builds at all, and is probably just me not knowing how to specify all the same options on meson as on autotools.

This is the error in question, again probably caused by the path options ikn my prior autotools build:
`
ERROR: Destination '/usr/share/help/ar/eom/figures/eom_save_as_window.png' already exists and is not a symlink
`

**Comment by raveit65** [comment]
> Building, installing, and uninstalling with EOM not previouslly installed went fine. If EOM built w autotools is already installed, we get an install error on replacing a file with a symlink, but this may just be meson building with different path options (I have a lot of specification for path in my autotools builds for multilib etc)..
> 
> That won't affect distro builds at all, and is probably just me not knowing how to specify all the same options on meson as on autotools.
> 
> This is the error in question, again probably caused by the path options ikn my prior autotools build: `ERROR: Destination '/usr/share/help/ar/eom/figures/eom_save_as_window.png' already exists and is not a symlink`

I think the problem is in our code. The images are there in this language https://github.com/mate-desktop/eom/tree/master/help/ar/figures
All other languages don't have this folder with images.
Only the C folder should have this images, ihmo.
I guess this folder can be removed safely.

**Review by zhuyaliang — COMMENTED** [review]
Prepare for testing

**Comment by zhuyaliang** [comment]
@raveit65 @whs These ```png``` files are useless files,I think we should first remove these files and then merge this PR，This requires modifying the ```help/meson.build``` file again

**Comment by whs** [comment]
It seems that the images are for language-specific application images? Although I can't open several of them, but at least ko/figures/eom_toolbar_editor_window.png seems to be the valid Korean version of the image.

**Comment by whs** [comment]
Removed all help images that are not openable by my local eom. It seems that meson build script do not directly reference them so no changes needed to meson.build

**Comment by whs** [comment]
Seems like if `symlink_media: false`, then localized images are supported, but meson will duplicate the image to each locale if localized images are not provided.



but if left as default (true), then every images are symlinked from C, so the localized images are ignored:



Which way should I pick?

(meson 0.61.2 Ubuntu 22.04)

**Comment by zhuyaliang** [comment]
My suggestion is
>but if left as default (true), then every images are symlinked from C, so the localized images are ignored

**Comment by whs** [comment]
Looking at the PR I think no further changes are needed then as that is already the current behavior

**Comment by zhuyaliang** [comment]
Thank you for your work. I will test it later

**Review by zhuyaliang — APPROVED** [review]
Agree to merge

**Pull request merged**